File: src/main/java/com/refinedmods/refinedstorage/container/slot/legacy/LegacyBaseSlot.java
package com.refinedmods.refinedstorage.container.slot.legacy;

import net.minecraft.inventory.IInventory;
import net.minecraft.inventory.container.Slot;

import java.util.function.BooleanSupplier;

public class LegacyBaseSlot extends Slot {
    private BooleanSupplier enableHandler = () -> true;

    public LegacyBaseSlot(IInventory inventory, int inventoryIndex, int x, int y) {
        super(inventory, inventoryIndex, x, y);
    }

    public LegacyBaseSlot setEnableHandler(BooleanSupplier enableHandler) {
        this.enableHandler = enableHandler;

        return this;
    }

    @Override
    public boolean isActive() {
        return enableHandler.getAsBoolean();
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/slot/legacy/LegacyBaseSlot.kt
package com.refinedmods.refinedstorage.container.slot.legacy

import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import net.minecraft.inventory.IInventory
import net.minecraft.inventory.container.Slot
import java.util.function.Supplier

open class LegacyBaseSlot(inventory: IInventory?, inventoryIndex: Int, x: Int, y: Int) : Slot(inventory, inventoryIndex, x, y) {
    private var enableHandler = Supplier { true }
    fun setEnableHandler(enableHandler: Supplier<Boolean>): LegacyBaseSlot {
        this.enableHandler = enableHandler
        return this
    }

    val isEnabled: Boolean
        get() = enableHandler.get()
}